lib/optimist-config-file.js

```diff
@@ -30,6 +31,9 @@ function parseConfigFile(configFileName) {
 	// parse it
 	try {
 		raw = fs.readFileSync(configFileName).toString();
+		if (replacementVars instanceof Object) {
+			raw = replace(raw, replacementVars);
+		}
```

Comment:
I'd move this block to YAML switch block - env variables replacement is going to work for YAML files only.